src/kw/kw.cpp

```diff
@@ -173,3 +240,77 @@ string KanoWorld::whoami(void)
 {
     return string("");
 }
+
+
+/**
+ *   Collects and returns the token from the Server Response data
+ */
+string KanoWorld::get_token(void)
+{
+    JSON_Value *schema = json_parse_string(server_response.c_str());
+    if (schema) {
+        token = string(json_object_dotget_string(json_object(schema), "data.token"));
+    }
+
+    return token;
+}
+
+
+/**
+ *   Collects and returns the token expiration date, "duration" field,
+ *   from the Server Response data.
+ */
+string KanoWorld::get_expiration_date(void)
+{
+    JSON_Value *schema = json_parse_string(server_response.c_str());
+    if (schema) {
+
+        // For some reason, the Unix time returned by the server is divide by 1000
+        // So we convert it back into a Unix time here. See:
+        // https://github.com/KanoComputing/kes-world-api/blob/develop/src/controllers/accounts.js#L35
+        long conversion = 1000;
+        string translate = string(json_object_dotget_string(json_object(schema), "data.duration"));
+        expiration_date = string( std::to_string (std::stol(translate.c_str()) * conversion) );
+    }
+
+    return expiration_date;
+}
+
+
+/**
+ *   Loads the data from the cached file.
+ */
+bool KanoWorld::load_data(void)
+{
+    JSON_Value *user_data = json_parse_file(data_filename.c_str());
+    if (!user_data) {
+        return false;
+    }
+
+    token = json_object_get_string(json_object(user_data), "token");
+    expiration_date = json_object_get_string(json_object(user_data), "duration");
+
+    return true;
+}
+
+
+/**
+ *   Saves the server response data fields into the cache local file.
+ */
+bool KanoWorld::save_data(void)
+{
+    JSON_Status rc;
+    JSON_Value *user_data = json_value_init_object();
+
+    if (!user_data) {
+        return false;
+    }
+
+    json_object_set_string(json_object(user_data), "token", get_token().c_str());
+    json_object_set_string(json_object(user_data), "duration", get_expiration_date().c_str());
+
+    rc = json_serialize_to_file(user_data, data_filename.c_str());
+    json_value_free(user_data);
+
+    return (rc == JSONSuccess);
```

Comment:
No need for parentheses:

```suggestion
    return rc == JSONSuccess;
```